src/main/java/com/sigmaukraine/messenger/controller/ChatController.java

```diff
@@ -0,0 +1,137 @@
+package com.sigmaukraine.messenger.controller;
+
+import com.sigmaukraine.messenger.domain.Chat;
+import com.sigmaukraine.messenger.repository.ChatRepository;
+import com.sigmaukraine.messenger.repository.SubjectRepository;
+import com.sigmaukraine.messenger.repository.UserRepository;
+import com.sigmaukraine.messenger.validation.ChatValidator;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.core.userdetails.User;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.validation.BindingResult;
+import org.springframework.web.bind.annotation.ModelAttribute;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+
+import java.util.List;
+
+@Controller
+@RequestMapping(value = "/subjects/{subjectId}/chats")
+public class ChatController {
+
+    private ChatRepository chatRepository;
+    private UserRepository userRepository;
+    private ChatValidator chatValidator;
+    private SubjectRepository subjectRepository;
+
+    public ChatController() {
+    }
+
+    @Autowired
+    public ChatController(ChatRepository chatRepository, UserRepository userRepository,
+                          ChatValidator chatValidator, SubjectRepository subjectRepository) {
+        this.chatRepository = chatRepository;
+        this.userRepository = userRepository;
+        this.chatValidator = chatValidator;
+        this.subjectRepository = subjectRepository;
+    }
+
+    @RequestMapping(value = "", method = RequestMethod.GET)
+    @PreAuthorize("isAuthenticated()")
+    public String list(Model model, @PathVariable Integer subjectId) {
+        List<Chat> chats= this.chatRepository.getListChatsBySubjectId(subjectId);
+        User userDetails = (User) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
+        com.sigmaukraine.messenger.domain.User user = userRepository.getUserByLogin(userDetails.getUsername());
```

Comment:
actually you can do this once (on login) and store the user object in it's session and access it whenever you want from both `JSP` and `HttpServletRequest` (but don't forget to invalidate that stored object if you update a record in database)